Odata-docs/odatalib/v7/using-ut8jsonwriter-for-better-performance.md

```diff
@@ -0,0 +1,58 @@
+---
+title: "Using Utf8JsonWriter for better performance"
+description: "This section describes how to configer the OData writer to use a Utf8JsonWriter-based writer for better performance."
+author: habbes
+ms.author: clhabins
+ms.date: 7/27/2022
+ms.topic: article
+ 
+---
+
+# Using Utf8JsonWriter for better performance
+
+**Applies To**: [!INCLUDE[appliesto-odataclient](../../includes/appliesto-odatalib-v7.md)]
+
+The `ODataMessageWriter` internally uses an implementation of [`IJsonWriter`](/dotnet/api/microsoft.odata.json.ijsonwriter) (and [`IJsonWriterAsync`](/dotnet/api/microsoft.odata.json.ijsonwriterasync?view=odata-core-7.0)) to write JSON output to the destination stream. This is a low-level JSON writer that exposes methods for writing individual properties and values. Before v7.13.0, `Microsoft.OData.Core` shipped a single default implementation (internally called `JsonWriter`) that can be constructed using the [`DefaultJsonWriterFactory`](/dotnet/api/microsoft.odata.json.defaultjsonwriterfactory).
+
+It's possible to replace the writer with a custom implementation by injecting your own implementation of [`IJsonWriterFactory`](/dotnet/api/microsoft.odata.json.ijsonwriterfactory) into the dependency-injection container. For more information about configuring dependency injection in ODate core library, [see this article](/odata/odatalib/di-support).
+
+Microsoft.OData.Core 7.13.0 introduced a new implementation of `IJsonWriter` and `IJsonWriterAsync` called `ODataUtf8JsonWriter`. This is based on .NET's built-in [Utf8JsonWriter](/dotnet/api/system.text.json.utf8jsonwriter). This writer is faster and uses less memory than the default `JsonWriter`.
+
+Since the existing `IJsonWriterFactory` is tightly coupled to `TextWriter`, we introduced a new interface `IStreamBasedJsonWriterFactory` which accepts the destination `Stream` as input instead of `TextWriter`. The default implementation of this interface is `DefaultStreamBasedJsonWriterFactory`, which creates instances of `ODataUtf8JsonWriter`.
+
+Therefore, in order to use `ODataUtf8JsonWriter`, you have to inject an instance of `DefaultStreamBasedJsonWriterFactory` to the dependency-injection container.
+
+Here's an example:
+
+```csharp
+using Microsoft.OData;
+using Microsoft.OData.Json;
+```
+
+```c#
+IContainerBuilder builder = new TestContainerBuilder();
+builder.AddDefaultODataServices();
+builder.AddService<IStreamBasedJsonWriterFactory>(sp => DefaultStreamBasedJsonWriterFactory.Default);
+```
+
+## Limitations and other considerations
```

Comment:
nit: I would Arrange the sub sections to be as follows as I believe the most pressing concern will be the .Net versions the other two I am not sure what the order will be.
```
Supported .Net versions
No Support for streaming writes
Choosing a JavaScriptEncoder
```